src/hooks/useTrades.ts

```diff
@@ -19,21 +25,69 @@ export function useTrades(): Trade[] {
     },
     dispatch,
   ] = useGlobalState()
-  const { userAddress, networkId, blockNumber } = useWalletConnection()
+  const { userAddress, networkId } = useWalletConnection()
+
+  const intervalId = useRef<null | NodeJS.Timeout>(null)
 
   useEffect(() => {
-    // Check only up to `latest - CONFIRMATION_BLOCKS` to reduce chance of re-orgs
-    const toBlock = blockNumber && blockNumber - CONFIRMATION_BLOCKS
-
-    if (userAddress && networkId && toBlock && (!lastCheckedBlock || lastCheckedBlock < toBlock)) {
-      getTrades({
-        userAddress,
-        networkId,
-        fromBlock: lastCheckedBlock,
-        toBlock,
-      }).then(newTrades => dispatch(appendTrades(newTrades, toBlock)))
+    function updateTrades(): void {
+      console.log(`Updating trades. Last checked block [${lastCheckedBlock}]`)
+      if (userAddress && networkId) {
+        // Don't want to update on every block
+        // So instead, we get the latest block when the time comes
+        web3.eth.getBlock('latest').then(({ number: blockNumber }) => {
+          console.log(`Updating trades. Latest block [${blockNumber}]`)
+
+          getTrades({
+            userAddress,
+            networkId,
+            // fromBlock is inclusive. If set, add 1 to avoid duplicates, otherwise return undefined
+            fromBlock: !lastCheckedBlock ? lastCheckedBlock : lastCheckedBlock + 1,
+            toBlock: blockNumber,
+          }).then(newTrades =>
+            dispatch(newTrades.length > 0 ? appendTrades(newTrades, blockNumber) : updateLastCheckedBlock(blockNumber)),
+          )
+        })
+      }
+    }
+
+    function clear(): void {
+      if (intervalId.current) clearInterval(intervalId.current)
+      intervalId.current = null
     }
-  }, [userAddress, networkId, blockNumber, lastCheckedBlock, dispatch])
+
+    clear()
```

Comment:
There's no need to call `clear` here, since interval is always cleared in clean-up phase of `useEffect()`
At this point there's no valid intervalId to clear